### PR TITLE
[OIS] Set of improvements for CI performance #2

### PR DESCRIPTION
**Pull request by ATmobica**
Improvements:
 - set maximum baudrate as default and adjust this to specific Matter example
 - remove the redundant `NDEBUG` setting in the config cmake 

Fixing:
#32

### CODE REVIEW

File: .github/workflows/examples-openiotsdk.yaml [hunk 1]
```diff
@@ -46,10 +46,11 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
+                testItem: ["test0", "test1", "test2", "test3", "test4", "test5"]
                 socketAPI: ["iotsocket", "lwip"]
                 
-        name: Open IoT SDK examples building
-        timeout-minutes: 140
+        name: Open IoT SDK examples building and testing
+        #timeout-minutes: 140
```
Comment (pan-) [inline .github/workflows/examples-openiotsdk.yaml:53]: 140 minutes is not enough ? 
Why not raise the timeout ?

File: config/openiotsdk/CMakeLists.txt [hunk 2]
```diff
@@ -149,7 +149,7 @@ else()
     set(CHIP_PROJECT_CONFIG "")
 endif()
 
-if (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+if (NOT ${CMAKE_BUILD_TYPE} STREQUAL "Release")
```
Comment (pan-) [inline config/openiotsdk/CMakeLists.txt:152]: Why change the test ? There's 4 default config: `Debug, Release, RelWithDebInfo and MinSizeRel`. 

The only debugging one is `Debug`.

File: config/openiotsdk/CMakeLists.txt [hunk 3]
```diff
@@ -292,9 +292,3 @@ endif()
 target_include_directories(openiotsdk-chip INTERFACE
     ${CHIP_ROOT}/config/openiotsdk/mbedtls
 )
-
-if (NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
```
Comment (pan-) [inline config/openiotsdk/CMakeLists.txt:296]: Is the goal to keep assertions in release builds ?

File: examples/shell/openiotsdk/CMakeLists.txt [hunk 4]
```diff
@@ -59,6 +59,11 @@ include(chip)
 
 add_subdirectory(${OPEN_IOT_SDK_EXAMPLE_COMMON}/app ./app_build)
 
+target_compile_definitions(openiotsdk-app
+    PRIVATE
+        IOT_SDK_APP_SERIAL_BAUDRATE=115200
```
Comment (pan-) [inline examples/shell/openiotsdk/CMakeLists.txt:64]: We can probably lower it, while working on another piece of code for the SDK, I discovered it was difficult to not drop characters in input if baudrate > 9600.

File: src/test_driver/openiotsdk/unit-tests/main/main.cpp [hunk 5]
```diff
@@ -36,25 +36,25 @@ static void test_thread(void * argument)
     int status;
     CHIP_ERROR err;
 
-    ChipLogAutomation("Open IoT SDK unit-tests start");
+    printf("Open IoT SDK unit-tests start\r\n");
```
Comment (pan-) [inline src/test_driver/openiotsdk/unit-tests/main/main.cpp:39]: Why do we move away from `ChipLogAutomation` to use `printf` instead ?

File: .github/workflows/examples-openiotsdk.yaml (hunk 1 above)
Comment (ATmobica) [reply .github/workflows/examples-openiotsdk.yaml:53]: I commented on it for testing purposes. Need to be restored. However, the timeouts need to be adjusted, according to the test results. I'll change them.

File: config/openiotsdk/CMakeLists.txt (hunk 2 above)
Comment (ATmobica) [reply config/openiotsdk/CMakeLists.txt:152]: Remnants of tests, to be turned over.

File: config/openiotsdk/CMakeLists.txt (hunk 3 above)
Comment (ATmobica) [reply config/openiotsdk/CMakeLists.txt:296]: The SDK release build provides this definition. We don't need it anymore.

File: examples/shell/openiotsdk/CMakeLists.txt (hunk 4 above)
Comment (ATmobica) [reply examples/shell/openiotsdk/CMakeLists.txt:64]: I don't see any issues with dropping characters in CI now. It seems that this baud rate is good enough for the shell example.

File: src/test_driver/openiotsdk/unit-tests/main/main.cpp (hunk 5 above)
Comment (ATmobica) [reply src/test_driver/openiotsdk/unit-tests/main/main.cpp:39]: In the shell unit test, we modify a bit the CHIP logger. It seems that this caused the line-splitting issue. I tried to use the print directly and it looks like it fixed the issue.

Comment (pan-) [reply src/test_driver/openiotsdk/unit-tests/main/main.cpp:39]: Can you provide a link to the changes made to the logger in the test ?

File: examples/shell/openiotsdk/CMakeLists.txt (hunk 4 above)
Comment (ATmobica) [reply examples/shell/openiotsdk/CMakeLists.txt]: Sorry, you have right on this point. The error occurred in CI last time. I changed it to the proposed value and now It works.

**Review by pan- — APPROVED** [review]

**Pull request merged**